usecases/data_model_usecase.go

```diff
@@ -66,10 +74,10 @@ func addUnicityConstraintStatusToDataModel(dataModel models.DataModel, uniqueInd
 				field.UnicityConstraint = models.ActiveUniqueConstraint
 			}
 			// cannot directly modify the struct field in the map, so we need to reassign it
-			dataModel.Tables[index.TableName].Fields[index.Fields[0]] = field
+			dm.Tables[index.TableName].Fields[index.Fields[0]] = field
```

Comment:
You could, with a deep copy.

Reply:
actually no: you really can't modify a struct that's in a map - it's not about how we copy, we really have to reassign the struct in the map

Reply:
because a struct kind of feels like a map (with predefined keys) but it really isn't